battleship_oo2_test.py

```diff
@@ -87,29 +78,35 @@ class Ship:
     """This class represents a ship, it keeps track if it is_sunk"""
     id_counter = 0
 
-    def __init__(self, list_tuples):
+    def __init__(self, iter_coordinates: 'Iterator[Coordinates]'):
         Ship.id_counter += 1
-        self.name = "Ship{}".format(Ship.id_counter)
-        self.cells = [Cell(Coordinates(*tuple_)) for tuple_ in list_tuples]
-
-    @property
-    def is_sunk(self):
-        return all(cell.is_hit for cell in self.cells)
-
-    def take_shot(self, shot):
-        for cell in self.cells:
-            if cell.coordinates == shot.coordinates:
-                cell.is_hit = True
-                print('{}, your shot hit {}!'.format(shot.shooter_name, self.name))
-                if self.is_sunk:
-                    print('{} sunk!'.format(self.name))
-                return True
+        self._name = "Ship{}".format(Ship.id_counter)
+        self._cells = [Cell(coordinates, status='Not Hit') for coordinates in iter_coordinates]
+        self._is_sunk = False
+
+    def update_is_sunk(self):
+        self._is_sunk = all(cell.get_status() == "Hit" for cell in self._cells)
+
+    def get_is_sunk(self):
+        return self._is_sunk
+
+    def get_name(self):
+        return self._name
+
+    def is_hit(self, shot: 'Shot'):
+        for cell in self._cells:
+            if cell.get_coordinates() == shot.coordinates:
+                if cell.get_status() == 'Not Hit':
+                    cell.set_status("Hit")
+                    self.update_is_sunk()
+                    return True
         return False
 
 
 class Board:
     """This class keeps track of the shots taken by a Player """
-    def __init__(self, board_len):
+
+    def __init__(self, board_len: int):
         self.coordinates = [Coordinates(x=x, y=y) for x in range(1, board_len) for y in range(1, board_len)]
```

Comment:
You populate coordinates but don't use it anywhere instead you use the shots_taken set.
Although this works with the current features, you should remove any unused code and I suggest you remove the shots_taken set and keep coordinates storing Cells instead of Coordinates. 

If the Board consists of Cells you can register a shot on the board by setting the status of the cell to something like 'Undiscovered' by default and make it 'Empty' or 'Hit' after a shot and it's also easy to verify if a shot is valid on the board or not.